### PR TITLE
Features/string includes test

### PR DESCRIPTION
**Pull request by ocampd1**
Created "counter" test on includes-method.

### CODE REVIEW

**Review by agamble33 — CHANGES_REQUESTED** [review]
Please review and delete space between variables and expect

File: src/test/javascript/stringExtensionSpec.js [hunk 1]
```diff
@@ -1,63 +1,108 @@
...
+define(['bytepushers'], function (BytePushers) {
+    describe("String extension tests:", function () {
 
         var testStr = "Hello this is a random string";
 
         describe('String.prototype.trim', function () {
-            it('will trim whitespace correctly', function() {
-                var result = "                       l     shdasdk;lg;al j          ".trim();
+            it('will trim whitespace correctly', function () {
+                var result = " happytime".trim();
```
Comment (tonte-pouncil) [inline src/test/javascript/stringExtensionSpec.js:7]: Please create stringUnderTest like so: `var stringUnderTest = " happytime";` 
And then use like so: `var actualResult = stringUnderTest.trim();`

File: src/test/javascript/stringExtensionSpec.js [hunk 2]
```diff
@@ -1,63 +1,108 @@
...
+
+                expect(result).toBe("happytime");
+            });
+            it("Will Not Trim in Front of a String", function () {
+                var frontTrimTest = "ABC";
```
Comment (tonte-pouncil) [inline src/test/javascript/stringExtensionSpec.js]: Rename `frontTrimTest` to `stringUnderTest`.

File: src/test/javascript/stringExtensionSpec.js [hunk 3]
```diff
@@ -1,63 +1,108 @@
...
+                var expectedResult = " ABC";
+                var actualResult = frontTrimTest.trim();
+
+                expect(expectedResult).not.toBe(actualResult);
+                //expect(false).not.toBe(true);
+            });
+
+            it("Will Trim in Front of a String", function () {
+                var frontTrimTest = " ABC";
```
Comment (tonte-pouncil) [inline src/test/javascript/stringExtensionSpec.js]: Rename `frontTrimTest` to `stringUnderTest`.

File: src/test/javascript/stringExtensionSpec.js [hunk 4]
```diff
@@ -1,63 +1,108 @@
...
+                var expectedResult = "ABC";
+                var actualResult = frontTrimTest.trim();
+
+                expect(expectedResult).toBe(actualResult);
+                //expect(false).not.toBe(true);
+            });
+
+            it("Will Not Trim", function () {
+                var noSpaceTrimTest = " ABC ";
```
Comment (tonte-pouncil) [inline src/test/javascript/stringExtensionSpec.js]: Rename `noSpaceTrimTest` to `stringUnderTest`.

File: src/test/javascript/stringExtensionSpec.js [hunk 5]
```diff
@@ -1,63 +1,108 @@
...
+                var actualResult = noSpaceTrimTest.trim();
+                expect(actualResult).toBe(expectedResult);
 
-                expect(result).toBe("l     shdasdk;lg;al j");
             });
         });
 
         describe('String.prototype.includes', function () {
-            it('can tell if a string includes a string', function() {
+            it('can tell if a string includes a string', function () {
```
Comment (tonte-pouncil) [inline src/test/javascript/stringExtensionSpec.js:39]: Remove space between function and open parentheses.

File: src/test/javascript/stringExtensionSpec.js [hunk 6]
```diff
@@ -1,63 +1,108 @@
...
                 var result = testStr.includes("o this i");
 
                 expect(result).toBe(true);
             });
 
-            it('can tell if a string does not include a string', function() {
+            it('can tell if a string does not include a string', function () {
```
Comment (tonte-pouncil) [inline src/test/javascript/stringExtensionSpec.js:45]: Remove space between function and open parentheses.

File: src/test/javascript/stringExtensionSpec.js [hunk 7]
```diff
@@ -1,63 +1,108 @@
...
 
                 expect(result).toBe(false);
             });
-        });
 
-        describe('String.prototype.toCamelCase', function () {
-            it('can format a sentence into camel case', function() {
-                var result = "Hello this is a sentence.".toCamelCase();
+            it('does this string include counter', function () {
+                var result = testStr.includes("counter");
```
Comment (tonte-pouncil) [inline src/test/javascript/stringExtensionSpec.js]: Set up the normal stringUnderTest, actualResult, and expectedResult variables.

File: src/test/javascript/stringExtensionSpec.js [hunk 8]
```diff
@@ -49,8 +49,8 @@ define(['bytepushers'], function (BytePushers) {
                 expect(result).toBe(false);
             });
 
-            it('does this string include play', function () {
-                var result = testStr.includes("play");
+            it('does this string include counter', function () {
+                var result = testStr.includes("counter");
```
Comment (agamble33) [inline src/test/javascript/stringExtensionSpec.js]: The referenced string could be rewritten to make more understandable or add a comment on the unit test.

**Review by agamble33 — CHANGES_REQUESTED** [review]
The referenced string could be rewritten to make more understandable or add a comment on the unit test.

File: src/test/javascript/stringExtensionSpec.js [hunk 9]
```diff
@@ -1,63 +1,108 @@
...
             });
-        });
 
-        describe('String.prototype.toNormalCase', function () {
-            it('can format a sentence into camel case', function() {
-                var result = "Hello world".toCamelCase().toNormalCase();
-                // helloWorld => "Hello World"
-                expect(result).toBe("Hello World");
+            it('Does the string include a space', function () {
+                var result = testStr.includes(" ");
```
Comment (tonte-pouncil) [inline src/test/javascript/stringExtensionSpec.js]: Set up the normal stringUnderTest, actualResult, and expectedResult variables.

File: src/test/javascript/stringExtensionSpec.js [hunk 10]
```diff
@@ -1,63 +1,108 @@
...
 
-            it('can format a sentence into camel case with a single letter word', function() {
-                var result = "Hello this is a sentence".toCamelCase().toNormalCase();
-                // Would think it would do this
-                // Hello this is a sentence => helloThisIsASentence => Hello This Is A Sentence
-                // Actually does this
-                // Hello this is a sentence => helloThisIsASentence => Hello This Is ASentence
-                expect(result).toBe("Hello This Is A Sentence");
+            it("", function () {
+                var result = testStr.includes("");
```
Comment (tonte-pouncil) [inline src/test/javascript/stringExtensionSpec.js]: Set up the normal stringUnderTest, actualResult, and expectedResult variables.

File: src/test/javascript/stringExtensionSpec.js [hunk 11]
```diff
@@ -1,63 +1,108 @@
...
             });
 
-            it('can format a sentence into camel case with a single letter word', function() {
-                var result = "Hello this is a sentence".toCamelCase().toNormalCase();
-                // Would think it would do this
-                // Hello this is a sentence => helloThisIsASentence => Hello This Is A Sentence
-                // Actually does this
-                // Hello this is a sentence => helloThisIsASentence => Hello This Is ASentence
-                expect(result).toBe("Hello This Is A Sentence");
+            it("", function () {
```
Comment (tonte-pouncil) [inline src/test/javascript/stringExtensionSpec.js]: Please provide a good test description.

File: src/test/javascript/stringExtensionSpec.js [hunk 12]
```diff
@@ -1,63 +1,108 @@
...
+
             });
-        });
 
-        describe('String.format', function () {
-            it('can format a string properly', function() {
-                var result = String.format("Hello {1} is {2} random {3}", "this", "a", "string");
+            describe('String.prototype.toCamelCase', function () {
+                it('can format a sentence into camel case', function () {
+                    var result = "Hello this is a sentence.".toCamelCase();
```
Comment (tonte-pouncil) [inline src/test/javascript/stringExtensionSpec.js]: Set up the normal stringUnderTest, actualResult, and expectedResult variables.

File: src/test/javascript/stringExtensionSpec.js [hunk 13]
```diff
@@ -1,63 +1,108 @@
...
+
+                    expect(result).toBe("helloThisIsASentence.");
+                });
+            });
+
+            describe('String.prototype.toNormalCase', function () {
+                it('can format a sentence into camel case', function () {
+                    var result = "Hello world".toCamelCase().toNormalCase();
```
Comment (tonte-pouncil) [inline src/test/javascript/stringExtensionSpec.js]: Set up the normal stringUnderTest, actualResult, and expectedResult variables.

File: src/test/javascript/stringExtensionSpec.js [hunk 14]
```diff
@@ -1,63 +1,108 @@
...
+                    // helloWorld => "Hello World"
+                    expect(result).toBe("Hello World");
+                });
+
 
-                expect(result).toBe(testStr); // testStr = "Hello this is a random string";
+                it('can format a sentence into camel case with a single letter word', function () {
+                    var result = "Hello this is a sentence".toCamelCase().toNormalCase();
```
Comment (tonte-pouncil) [inline src/test/javascript/stringExtensionSpec.js]: Set up the normal stringUnderTest, actualResult, and expectedResult variables.

File: src/test/javascript/stringExtensionSpec.js [hunk 15]
```diff
@@ -1,63 +1,108 @@
...
+                    // Would think it would do this
+                    // Hello this is a sentence => helloThisIsASentence => Hello This Is A Sentence
+                    // Actually does this
+                    // Hello this is a sentence => helloThisIsASentence => Hello This Is ASentence
+                    expect(result).toBe("Hello This Is A Sentence");
+                });
+            });
+
+            describe('String.format', function () {
+                it('can format a string properly', function () {
```
Comment (tonte-pouncil) [inline src/test/javascript/stringExtensionSpec.js]: Set up the normal stringUnderTest, actualResult, and expectedResult variables.

**Review by tonte-pouncil — CHANGES_REQUESTED** [review]

File: src/test/javascript/stringExtensionSpec.js (hunk 1 above)
Comment (agamble33) [reply src/test/javascript/stringExtensionSpec.js:7]: updated

File: src/test/javascript/stringExtensionSpec.js (hunk 2 above)
Comment (agamble33) [reply src/test/javascript/stringExtensionSpec.js]: done

File: src/test/javascript/stringExtensionSpec.js (hunk 3 above)
Comment (agamble33) [reply src/test/javascript/stringExtensionSpec.js]: done

File: src/test/javascript/stringExtensionSpec.js (hunk 4 above)
Comment (agamble33) [reply src/test/javascript/stringExtensionSpec.js]: done

File: src/test/javascript/stringExtensionSpec.js (hunk 5 above)
Comment (agamble33) [reply src/test/javascript/stringExtensionSpec.js:39]: reformat keeps space

File: src/test/javascript/stringExtensionSpec.js (hunk 6 above)
Comment (agamble33) [reply src/test/javascript/stringExtensionSpec.js:45]: done

File: src/test/javascript/stringExtensionSpec.js (hunk 11 above)
Comment (agamble33) [reply src/test/javascript/stringExtensionSpec.js]: done updated

File: src/test/javascript/stringExtensionSpec.js (hunk 9 above)
Comment (agamble33) [reply src/test/javascript/stringExtensionSpec.js]: done

File: src/test/javascript/stringExtensionSpec.js (hunk 10 above)
Comment (agamble33) [reply src/test/javascript/stringExtensionSpec.js]: done

File: src/test/javascript/stringExtensionSpec.js (hunk 12 above)
Comment (agamble33) [reply src/test/javascript/stringExtensionSpec.js]: done

File: src/test/javascript/stringExtensionSpec.js (hunk 13 above)
Comment (agamble33) [reply src/test/javascript/stringExtensionSpec.js]: done

File: src/test/javascript/stringExtensionSpec.js (hunk 14 above)
Comment (agamble33) [reply src/test/javascript/stringExtensionSpec.js]: done

File: src/test/javascript/stringExtensionSpec.js (hunk 15 above)
Comment (agamble33) [reply src/test/javascript/stringExtensionSpec.js]: done

**Review by agamble33 — APPROVED** [review]

**Review by tonte-pouncil — APPROVED** [review]

**Pull request closed without merging**